content/docs/projects/trust-manager.md

```diff
@@ -56,79 +70,299 @@ spec:
+is also written to the `status` field on any synced Bundle resource which uses the default CA package.
+
+## Upgrading A Default CA Package Using Helm
+
+Say we want to do an in-place upgrade of our default CA package to tagged version `XYZ` - without upgrading trust-manager.
+
+We'll assume the Helm release is called "trust-manager" and that we've installed into the `cert-manager` namespace.
+
+> ⚠️ This upgrade process assumes that it's the only thing running. If another user or process changes Helm values
+> while you're doing this process, you might overwrite their work.
+
+First, we'll dump our current Helm values, so we don't lose them:
+
 ```bash
-kubectl get cm -A --field-selector=metadata.name=example-bundle
-NAMESPACE            NAME             DATA   AGE
-cert-manager         example-bundle   1      2m18s
-default              example-bundle   1      2m18s
-kube-node-lease      example-bundle   1      2m18s
-kube-public          example-bundle   1      2m18s
-kube-system          example-bundle   1      2m18s
-local-path-storage   example-bundle   1      2m18s
+helm get values -n cert-manager trust-manager -oyaml > values.yaml
 ```
 
+Next, if `defaultPackageImage.tag` is already set in `values.yaml`, update it. Otherwise, add it.
+You can find the available tags [on `quay.io`](https://quay.io/repository/jetstack/cert-manager-package-debian?tab=tags&tag=latest).
+
+```yaml
+# values.yaml
+...
+defaultPackageImage:
+  tag: XYZ
+```
+
+Finally, apply back the changes, being sure to manually specify the version of trust-manager which is installed, to avoid
+also updating the trust-manager controller at the same as the default CA package:
+
 ```bash
-kubectl get cm -n kube-system example-bundle -o jsonpath="{.data['target-key']}"
-123
-ABC
-hello world!
+# Get the currently installed version. You could do this manually if you find that easier.
+TRUST_MANAGER_VER=$(helm list --filter "^trust-manager$" -n cert-manager -ojson | jq -r ".[0].app_version")
+
+# Check the version makes sense
+echo $TRUST_MANAGER_VER
+
+# Run the upgrade
+helm upgrade -f values.yaml -n cert-manager trust-manager jetstack/trust-manager --version $TRUST_MANAGER_VER
 ```
+
+## Avoiding Gotchas
+
+TLS is a complicated beast and there are plenty of ways to use TLS certificates in ways which will cause problems
+down the line. We'd like to document a few of these potential gotchas here.
+
+If you're planning on running trust-manager in production and you're using more than just the default CA package,
+we **strongly** advise you to read and understand this section. It could save you from causing an outage later.
+
+> ℹ️ These gotchas aren't specific to trust-manager and you could run into any of them with any method of managing TLS trust!
+
+### Bundling Intermediates
+
+A very common anti-pattern in TLS is that servers fail to send their chain along with their leaf certificate. The correct
```

Comment:
can we add some references here?

Reply:
I rewrote this; my "references" would be "bitter personal experience" so I don't really have any useful links.